types/prisma/schema.prisma

```diff
@@ -70,12 +70,14 @@ model TestCase {
 }
 
 model User {
-  utorid    String     @id
-  email     String
-  solutions Solution[]
-  roles     Role[]
-  admin     Boolean    @default(false)
-  TestCase  TestCase[]
+  utorid     String     @id
+  email      String
+  solutions  Solution[]
+  roles      Role[]
+  admin      Boolean    @default(false)
+  TestCase   TestCase[]
+  surname    String?
+  givenName  String?
```

Comment:
```suggestion
  surname    String
  givenName  String
```
If they are on every request dont make them optional. Note the email isnt optional either...

Reply:
fair. but is there any edgecase where a user with a mononym would only send one header? i don't want it to break for them